src/xrpld/app/tx/detail/DepositPreauth.cpp

```diff
@@ -33,45 +38,90 @@ DepositPreauth::preflight(PreflightContext const& ctx)
     if (!ctx.rules.enabled(featureDepositPreauth))
         return temDISABLED;
 
+    bool const authArrPresent = ctx.tx.isFieldPresent(sfAuthorizeCredentials);
+    bool const unauthArrPresent =
+        ctx.tx.isFieldPresent(sfUnauthorizeCredentials);
+    int const authCredPresent =
+        static_cast<int>(authArrPresent) + static_cast<int>(unauthArrPresent);
+
+    if (authCredPresent && !ctx.rules.enabled(featureCredentials))
+        return temDISABLED;
+
     if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
         return ret;
 
-    auto& tx = ctx.tx;
-    auto& j = ctx.j;
-
-    if (tx.getFlags() & tfUniversalMask)
+    if (ctx.tx.getFlags() & tfUniversalMask)
     {
-        JLOG(j.trace()) << "Malformed transaction: Invalid flags set.";
+        JLOG(ctx.j.trace()) << "Malformed transaction: Invalid flags set.";
         return temINVALID_FLAG;
     }
 
     auto const optAuth = ctx.tx[~sfAuthorize];
     auto const optUnauth = ctx.tx[~sfUnauthorize];
-    if (static_cast<bool>(optAuth) == static_cast<bool>(optUnauth))
+    int const authPresent = static_cast<int>(optAuth.has_value()) +
+        static_cast<int>(optUnauth.has_value());
+
+    if (authPresent + authCredPresent != 1)
     {
         // Either both fields are present or neither field is present.  In
         // either case the transaction is malformed.
-        JLOG(j.trace())
+        JLOG(ctx.j.trace())
             << "Malformed transaction: "
                "Invalid Authorize and Unauthorize field combination.";
         return temMALFORMED;
     }
 
     // Make sure that the passed account is valid.
-    AccountID const target{optAuth ? *optAuth : *optUnauth};
-    if (target == beast::zero)
+    if (authPresent)
     {
-        JLOG(j.trace()) << "Malformed transaction: Authorized or Unauthorized "
-                           "field zeroed.";
-        return temINVALID_ACCOUNT_ID;
-    }
+        AccountID const& target(optAuth ? *optAuth : *optUnauth);
+        if (!target)
+        {
+            JLOG(ctx.j.trace())
+                << "Malformed transaction: Authorized or Unauthorized "
+                   "field zeroed.";
+            return temINVALID_ACCOUNT_ID;
+        }
 
-    // An account may not preauthorize itself.
-    if (optAuth && (target == ctx.tx[sfAccount]))
+        // An account may not preauthorize itself.
+        if (target == ctx.tx[sfAccount])
```

Comment:
Why is the `optAuth &&` removed here?

Reply:
I had the same comment a couple of weeks ago. Somehow it got moved...

Reply:
As above - it is under  ` if (authPresent)` condition now.

Reply:
Reverted